Adds guidelines to get model in migration

gurkanindibay opened the pull request:
Detailed description to the problem that is mentioned in https://github.com/citusdata/django-multitenant/issues/138#issuecomment-1469497358
Added unit tests and documentation to guide the usage of models which are submodel of TenantModel

jonels-msft reviewed (approved):
I moved the new docs from a separate page into the usage section.

I also rewrote the migration docs to try to compare "before and after" more clearly. @gurkanindibay I think the PR is good to merge, unless my rewrite contains technical errors! Please double check.

gurkanindibay commented:
> I moved the new docs from a separate page into the usage section.
> 
> I also rewrote the migration docs to try to compare "before and after" more clearly. @gurkanindibay I think the PR is good to merge, unless my rewrite contains technical errors! Please double check.

It looks great. The meaning is now more apparent and easier to comprehend Thanks @jonels-msft
